src/controllers/user.ts

```diff
@@ -76,10 +81,16 @@ const userController = {
                 if (!user)
                     res.status(statusCodes.NOT_FOUND).json({ status: statusCodes.NOT_FOUND, message: "User not found" });
                 else {
+                    const foundByEmail = await userDBInteractions.findByEmail(req.body.email);
+                    const foundByUsername = await userDBInteractions.findByUsername(req.body.username);
+                    if (foundByEmail && !(foundByEmail._id.equals(user._id))) {
+                        return res.status(statusCodes.BAD_REQUEST).json({ status: statusCodes.BAD_REQUEST, message: "User with that email already exists" });
```

Comment:
should we use the 409 (conflict found) HTTP response code for these returns?

Reply:
Did not know 409 was a thing - thanks :)